Yoast/wordpress-seo
# Add error descriptions to the validated URL fields

## Summary

This PR can be summarized in the following changelog entry:

* Improves user input validation feedback and suggestions for error correction.

## Relevant technical choices:

- Removed a test that looked redundant to me, see https://github.com/Yoast/wordpress-seo/pull/13240/files#r316718706
- Adjusted another test following existing examples

## Test instructions
- on trunk
- go to SEO > Social > Accounts 
- fill all the fields with invalid data
- save
- after page reload, see that only the Twitter Username has an error description:

<img width="652" alt="Screenshot 2019-08-22 at 14 11 58" src="https://user-images.githubusercontent.com/1682452/63525688-61058600-c4fe-11e9-843a-bfba07a38bc2.png">

- on this branch
- repeat the steps above
- see that all the URL fields have an error description:

<img width="687" alt="Screenshot 2019-08-22 at 14 30 39" src="https://user-images.githubusercontent.com/1682452/63525726-71b5fc00-c4fe-11e9-9833-b5bf33925b55.png">


## UI changes

## Documentation

## Quality assurance


## Request for feedback
Ideally, the descriptions should provide _suggestions for correction_, which is a bit tricky for URLs.

1
Should the input name be specified in the description e.g. `...the format of the Facebook Page URL ...`, `... the format of the Instagram URL ...` or just use a generic string `Please check the format of the URL you entered.`?

The former option adds a few more strings to translate. The latter could be a bit confusing when there are multiple URL errors in the same page.

2
Any idea for better _suggestions for correction_ or better wording? I'm not sure we can programmatically identify _which part_ of the URL isn't well formed. Maybe we could provide a generic message "must start with `https://`, not contain spaces, blah blah" but that seems a bit overkill.

Fixes https://github.com/Yoast/bugreports/issues/552
